tests/unit/regression/GaussianProcessRegression_unittest.cc

```diff
@@ -198,16 +198,17 @@ TEST(GaussianProcessRegression, apply_regression_on_training_features)
 	auto liklihood=std::make_shared<GaussianLikelihood>(0.25);
 
 	// specify GP regression with exact inference
-	auto inf=std::make_shared<ExactInferenceMethod>(kernel, features_train,
```

Comment:
but shouldn't this constructor be changed so it works without labels?

Reply:
yes, the ctor should be modified eventually
@LiuYuHui you can decide whether you want to change it in this PR, or in subsequent ones?

Reply:
(I like smaller PRs more, so if you're ok doing this in another one, I will merge this one)

Reply:
Sounds good, we can go through this when we merge into develop